public/pages/Main/Main.tsx

```diff
@@ -32,62 +44,152 @@ enum Pathname {
-        availableConfigTypes: serverFeatures.availableConfigTypes,
-        tooltipSupport: serverFeatures.tooltipSupport,
+        dataSourceId,
+        dataSourceLabel,
+        dataSourceReadOnly,
+        dataSourceLoading,
+        availableChannels,
+        availableConfigTypes,
+        tooltipSupport
       });
     } else {
-      // Feature API call failed, allow all configs to avoid UI breaking.
-      // User requests will still be validated by backend.
+      const { dataSourceId = "", dataSourceLabel = "" } = this.state;
+      const dataSourceReadOnly = false;
+      const dataSourceLoading = this.props.multiDataSourceEnabled;
+      const defaultConfigTypes = [
+        'slack',
+        'chime',
+        'microsoft_teams',
+        'webhook',
+        'email',
+        'sns',
+        'smtp_account',
+        'ses_account',
+        'email_group',
+      ];
+
       this.setState({
-        availableChannels: CHANNEL_TYPE,
-        availableConfigTypes: [
-          'slack',
-          'chime',
-          'microsoft_teams',
-          'webhook',
-          'email',
-          'sns',
-          'smtp_account',
-          'ses_account',
-          'email_group',
-        ],
-        tooltipSupport: serverFeatures.tooltipSupport,
+        dataSourceId,
+        dataSourceLabel,
+        dataSourceReadOnly,
+        dataSourceLoading,
+        availableChannels: this.props.multiDataSourceEnabled ? CHANNEL_TYPE : defaultConfigTypes,
+        availableConfigTypes: defaultConfigTypes,
+        tooltipSupport: false
       });
     }
   }
 
-  render() {
+  onSelectedDataSources = (dataSources: DataSourceOption[]) => {
+    const { id = "", label = "" } = dataSources[0] || {};
+    if (this.state.dataSourceId !== id || this.state.dataSourceLabel !==label) {
+      this.setState({
+        dataSourceId: id,
+        dataSourceLabel: label,
+      });
+    }
+    if (this.state.dataSourceLoading) {
+      this.setState({
+        dataSourceLoading: false,
+      });
+    }
+  };
+
+  getServices(http: HttpSetup) {
     const {
       location: { pathname },
     } = this.props;
+    let notificationService;
+    if (this.props.multiDataSourceEnabled) {
+      notificationService = new NotificationService(http, this.state.dataSourceId, this.props.multiDataSourceEnabled);
+    }
+    else {
+      notificationService = new NotificationService(http);
+    }
+    const services = {
+      notificationService,
+    };
+    return services;
+  }
 
+  render() {
+    const {
+      location: { pathname },
+    } = this.props;
+    let DataSourceMenuSelectable, DataSourceMenuView;
+    let activeOption: DataSourceOption[] | undefined;
+    if (this.props.multiDataSourceEnabled) {
+      DataSourceMenuSelectable = this.props.dataSourceManagement?.ui?.getDataSourceMenu<DataSourceSelectableConfig>();
+      DataSourceMenuView = this.props.dataSourceManagement.ui.getDataSourceMenu<DataSourceViewConfig>();
```

Comment:
looks like we expect dataSourceManagement to exist in props in line 183 but not 100% at line 182. Is dataSourceManagement optional or do we know for sure it exists if this.props.multiDataSourceEnabled is true?

Reply:
good catch, addressed